torchgeo/datasets/mmearth.py

```diff
@@ -0,0 +1,594 @@
+            'B9',
+            'B10',
+            'B11',
+            'B12',
+        ],
+        'sentinel2_cloudmask': ['QA60'],
+        'sentinel2_cloudprod': ['MSK_CLDPRB'],
+        'sentinel2_scl': ['SCL'],
+        'sentinel1_asc': ['VV', 'VH', 'HH', 'HV'],
+        'sentinel1_desc': ['VV', 'VH', 'HH', 'HV'],
+        'aster': ['elevation', 'slope'],
+        'era5': [
+            'prev_month_avg_temp',
+            'prev_month_min_temp',
+            'prev_month_max_temp',
+            'prev_month_total_precip',
+            'curr_month_avg_temp',
+            'curr_month_min_temp',
+            'curr_month_max_temp',
+            'curr_month_total_precip',
+            'year_avg_temp',
+            'year_min_temp',
+            'year_max_temp',
+            'year_total_precip',
+        ],
+        'dynamic_world': ['landcover'],
+        'canopy_height_eth': ['height', 'std'],
+        'lat': ['sin', 'cos'],
+        'lon': ['sin', 'cos'],
+        'biome': ['biome'],
+        'eco_region': ['eco_region'],
+        'month': ['sin_month', 'cos_month'],
+        'esa_worldcover': ['map'],
+    }
+
+    # See https://github.com/vishalned/MMEarth-train/blob/8d6114e8e3ccb5ca5d98858e742dac24350b64fd/MODALITIES.py#L36
+    no_data_vals: ClassVar[dict[str, int | float]] = {
+        'sentinel2': 0,
+        'sentinel2_cloudmask': 65535,
+        'sentinel2_cloudprod': 65535,
+        'sentinel2_scl': 255,
+        'sentinel1_asc': float('-inf'),
+        'sentinel1_desc': float('-inf'),
+        'aster': float('-inf'),
+        'canopy_height_eth': 255,
+        'dynamic_world': 0,
+        'esa_worldcover': 255,
+        'lat': float('-inf'),
+        'lon': float('-inf'),
+        'month': float('-inf'),
+        'era5': float('inf'),
+        'biome': 255,
+        'eco_region': 65535,
+    }
+
+    norm_modes: ClassVar[list[str]] = ['z-score', 'min-max']
+
+    modality_category_name: ClassVar[dict[str, str]] = {
+        'sentinel1': 'image_',
+        'sentinel2': 'image_',
+        'sentinel2_cloudmask': 'mask_',
+        'sentinel2_cloudprod': 'mask_',
+        'sentinel2_scl': 'mask_',
+        'aster': 'image_',
+        'era5': 'image_',
+        'canopy_height_eth': 'image_',
+        'dynamic_world': 'mask_',
+        'esa_worldcover': 'mask_',
+    }
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        subset: str = 'MMEarth',
+        modalities: Sequence[str] = all_modalities,
+        modality_bands: dict[str, list[str]] | None = None,
+        normalization_mode: str = 'z-score',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+    ) -> None:
+        """Initialize the MMEarth dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            subset: one of "MMEarth", "MMEarth64", or "MMEarth100k"
+            modalities: list of modalities to load
+            modality_bands: dictionary of modality bands, see `all_modality_bands`
```

Comment:
At the moment, `all_modality_bands` isn't documented. Should it be?

Reply:
I meant to point to the class variable `all_modality_bands`

Reply:
mhm, I thought I could have the reference to the existing class attributes, do I need to redefine them again?

Reply:
The class attributes exist in the code, but not in the documentation. To make them exist in the documentation too, you need to use `#:` to describe the attribute. You don't need to define them twice.

Reply:
Mhm, but it's all displayed in uppercase on the docs page, which will be misleading because that is not how the actual data is defined. Is there a way to control that?